src/shared/Core.Tests/StringExtensionsTests.cs

```diff
@@ -293,5 +293,14 @@ public void StringExtensions_TrimMiddle_String_ComparisonType(string input, stri
             string actual = StringExtensions.TrimMiddle(input, trim, comparisonType);
             Assert.Equal(expected, actual);
         }
+
+        [Theory]
+        [InlineData("FooBar", "foo_bar")]
+        [InlineData("FBBaz", "fb_baz")]
+        [InlineData("Foo", "foo")]
```

Comment:
```suggestion
        [InlineData("FooBar", "foo_bar")]
        [InlineData("fooBar", "foo_bar")]
        [InlineData("FBBaz", "fb_baz")]
        [InlineData("Foo", "foo")]
        [InlineData("Fo", "fo")]
        [InlineData("fO", "f_o")]
        [InlineData("OO", "oo")]
        [InlineData("F", "f")]
        [InlineData("", "")]
```

Here are some more test data to use. One of them reveals a bug 🐛